Dart/src/com/jetbrains/lang/dart/ide/runner/server/vmService/frame/DartVmServiceValue.java

```diff
@@ -320,30 +320,32 @@ private boolean computeListPresentation(@NotNull final XValueNode node) {
   }
 
   private void computeDefaultPresentation(@NotNull final XValueNode node) {
-    myDebugProcess.getVmServiceWrapper()
-      .evaluateInTargetContext(myIsolateId, myInstanceRef.getId(), "toString()", new VmServiceConsumers.EvaluateConsumerWrapper() {
-        @Override
-        public void received(final InstanceRef toStringInstanceRef) {
-          if (toStringInstanceRef.getKind() == InstanceKind.String) {
-            final String string = toStringInstanceRef.getValueAsString();
-            // default toString() implementation returns "Instance of 'ClassName'" - no interest to show
-            if (string.equals("Instance of '" + myInstanceRef.getClassRef().getName() + "'")) {
-              noGoodResult();
-            }
-            else {
-              node.setPresentation(getIcon(), myInstanceRef.getClassRef().getName(), string, true);
-            }
+    myDebugProcess.getVmServiceWrapper().callToString(myIsolateId, myInstanceRef.getId(), new VmServiceConsumers.InvokeConsumerWrapper() {
+      @Override
+      public void received(final InstanceRef toStringInstanceRef) {
+        if (toStringInstanceRef.getKind() == InstanceKind.String) {
+          String string = toStringInstanceRef.getValueAsString();
+          // default toString() implementation returns "Instance of 'ClassName'" - no interest to show
+          if (string.equals("Instance of '" + myInstanceRef.getClassRef().getName() + "'")) {
+            noGoodResult();
           }
           else {
-            noGoodResult(); // unlikely possible
+            if (toStringInstanceRef.getValueAsStringIsTruncated()) {
+              string += "...";
```

Comment:
I've changed this line to 
`addFullStringValueEvaluator(node, toStringInstanceRef);`
just like in other similar places. It adds `...View` link to see full value in a popup.